plugins/destination/gremlin/client/client.go

```diff
@@ -95,11 +96,25 @@ func (c *Client) getAuthInfo(ctx context.Context, baseURL string) (gremlingo.Aut
 		// emptyStringSHA256 is a SHA256 of an empty string
 		const emptyStringSHA256 = `e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`
 
-		req, err := http.NewRequest(http.MethodGet, baseURL, strings.NewReader(""))
+		urlToSign := baseURL
+		if c.spec.AWSNeptuneHost != "" {
+			u, err := url.Parse(baseURL)
+			if err != nil {
+				return nil, err
+			}
+			u.Host = c.spec.AWSNeptuneHost // signer.SignHTTP will use this as the Host header
```

Comment:
AWS Signer doesn't take into account if the passed-in `req.Header` has a `Host` field, it simply parses the URL and uses that. So we need to fake it.